config/routes.rb

```diff
@@ -1,6 +1,15 @@
+# frozen_string_literal: true
+
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+  # root 'root/index'
 
+  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+  namespace :api do
+    namespace :v1 do
+      resources :greetings, only: [:index]
+    end
+  end
   # Defines the root path route ("/")
+  root 'greetings#index'
   # root "articles#index"
```

Comment:
When I run your application I get a routing error.

![image](https://user-images.githubusercontent.com/38088818/197354189-21f9fbcc-aa96-4029-ac4e-039eb1ea36cb.png)

and on the console:

![Screenshot from 2022-10-22 19-31-11](https://user-images.githubusercontent.com/38088818/197354405-51b950c1-4359-43d0-b92b-617b5b4e74c4.png)

- This is because of the way you named your controller. Since your route is `root 'greetings#index'`, rails expects a controller and class named with the keyword `greetings`. In your case, your controller is named `root_controller.rb` instead of `greetings_controller.rb`.

and your class is `RootController` instead of `GreetingsController`

![image](https://user-images.githubusercontent.com/38088818/197355243-d9dbe051-6906-40f4-a121-a44e78f1d814.png)

Because of that, according to rails conventions, the `greetings controller` does not exist. Making the route `greetings#index` invalid.
**NOTE:** Remember that rails have `naming conventions`, therefore the names for models, controllers, classes, and routes for particular entities should follow the same naming convention.